extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -537,14 +534,6 @@ export class AmpStoryPage extends AMP.BaseElement {
     }
   }
 
-  /** @return {!Promise} */
-  beforeVisible() {
-    // Ensures a dynamically added page-attachment or page-outlink element is built.
-    // This happens by amp-story-ads.
-    this.renderOpenAttachmentUI_();
-    return this.maybeApplyFirstAnimationFrameOrFinish();
```

Comment:
cc @calebcordry to help testing this for Ads

Reply:
I tried following the logic and doesn't seem it would change anything: `beforeVisible` is called at the end of layouotCallback inside a `Promise.all` but immediately before that there's a `renderOpenAttachmentUI` so the two calls to `renderOpenAttachmentUI` happen synchronously with each other.

The other call to `beforeVisible` happens in layoutStory with just the first initial page of the story, which I don't think can eever be an ad.

Will still wait for @calebcordry to confirm this.

Reply:
Good catch! Matias and I just manually tested and confirmed this does break the new CTA rendering. I will look to see if I can have ads install the `page-outlink` extension instead of doing it in `beforeVisible`. If this works as expected it should be safe to delete here.